iml-wire-types/src/lib.rs

```diff
@@ -1294,6 +1336,85 @@ impl EndpointName for Log {
     }
 }
 
+// A log message from GraphQL endpoint
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[cfg_attr(feature = "graphql", derive(juniper::GraphQLObject))]
+pub struct LogMessage {
+    pub id: i32,
+    pub datetime: chrono::DateTime<Utc>,
+    pub facility: i32,
+    pub fqdn: String,
+    pub message: String,
+    pub message_class: MessageClass,
+    pub severity: LogSeverity,
+    pub tag: String,
+}
+
+impl TryFrom<LogMessageRecord> for LogMessage {
+    type Error = &'static str;
+
+    fn try_from(record: LogMessageRecord) -> Result<Self, Self::Error> {
+        Ok(Self {
+            id: record.id,
+            datetime: record.datetime,
+            facility: record.facility as i32,
+            fqdn: record.fqdn,
+            message: record.message,
+            message_class: record.message_class.try_into()?,
+            severity: record.severity.try_into()?,
+            tag: record.tag,
+        })
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[cfg_attr(feature = "graphql", derive(juniper::GraphQLEnum))]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum SortDir {
+    Asc,
+    Desc,
+}
+
+impl Default for SortDir {
+    fn default() -> Self {
+        Self::Asc
+    }
+}
+
+impl Deref for SortDir {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            Self::Asc => "ASC",
+            Self::Desc => "DESC",
+        }
+    }
+}
+
+pub mod logs {
+    use crate::LogMessage;
+
+    #[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+    #[cfg_attr(feature = "graphql", derive(juniper::GraphQLObject))]
+    pub struct Meta {
+        pub total_count: i32,
+    }
+
+    #[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+    #[cfg_attr(feature = "graphql", derive(juniper::GraphQLObject))]
+    pub struct LogResponse {
+        pub data: Vec<LogMessage>,
+        pub meta: Meta,
+    }
+
+    #[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+    #[cfg_attr(feature = "graphql", derive(juniper::GraphQLObject))]
+    pub struct Resp {
```

Comment:
At the very least, this should be in `iml-graphql-queries`.

Reply:
Moved